pkg/proxy/metaproxier/meta_proxier.go

```diff
@@ -62,33 +64,44 @@ func (proxier *metaProxier) SyncLoop() {
 
 // OnServiceAdd is called whenever creation of new service object is observed.
 func (proxier *metaProxier) OnServiceAdd(service *v1.Service) {
-	if *(service.Spec.IPFamily) == v1.IPv4Protocol {
-		proxier.ipv4Proxier.OnServiceAdd(service)
+	svcName := types.NamespacedName{Namespace: service.Namespace, Name: service.Name}
+	if utilproxy.ShouldSkipService(svcName, service) {
```

Comment:
Just make `ShouldSkipService` not take the `svcName` arg; it can just use `service.Namespace` and `service.Name` if it needs to log a message (which it normally doesn't). (And none of the other existing callers have a `NamespaceName` object in advance either.) (Also, userspace/proxier.go is redundantly re-logging the same message that `ShouldSkipService` logs, which you should fix too.) (So, one commit to fix the existing `ShouldSkipService` usage, then a second commit to use it in the metaproxier.)

Reply:
done